src/RelayClient.ts

```diff
@@ -254,55 +255,36 @@ class RelayClient extends EnvelopingEventEmitter {
 
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-ignore FIXME: remove the ts ignore once the method is consumed
-  private _prepareRelayHttpRequest = async (
-    { feesReceiver }: HubInfo,
-    { request, relayData }: RelayRequest
+  private _prepareHttpRequest = async (
+    relayRequest: RelayRequest
   ): Promise<EnvelopingTxRequest> => {
-    const { callForwarder } = relayData;
-    const { relayHub: relayHubAddress } = request;
-
-    if (!callForwarder) {
-      throw new Error('Call forwarder must be defined.');
-    }
-
-    if (
```

Comment:
I think that this validation should remain here and pass this value as parameter, since it would be coming from the server.